cura/Machines/Models/QualityManagementModel.py

```diff
@@ -233,6 +247,31 @@ def _qualityChangesListChanged(self, container: "ContainerInterface") -> None:
         if container.getMetaDataEntry("type") == "quality_changes":
             self._update()
 
+    @pyqtSlot("QVariantMap", result = str)
+    def getQualityItemDisplayName(self, quality_model_item: Dict[str, Any]) -> str:
+        display_name = quality_model_item["name"]
+
+        quality_group = quality_model_item["quality_group"]
+        is_read_only = quality_model_item["is_read_only"]
+        intent_category = quality_model_item["intent_category"]
+
+        # Not a custom quality
+        if is_read_only:
+            return display_name
+
+        # A custom quality
+        if intent_category != "default":
+            from cura.Machines.Models.IntentCategoryModel import IntentCategoryModel
+            intent_display_name = IntentCategoryModel.name_translation.get(intent_category,
```

Comment:
We don't need to do it like this, right? The catalog will already have the labels, so you should be able to just directly ask it (which prevents you from having to import the category model and ask that what the translation is)

Reply:
Fixed in 2033f24a1